samza-core/src/main/java/org/apache/samza/coordinator/MetadataResourceUtil.java

```diff
@@ -55,6 +55,7 @@ public MetadataResourceUtil(JobModel jobModel, MetricsRegistry metricsRegistry,
   public void createResources() {
     if (checkpointManager != null) {
       checkpointManager.createResources();
+      checkpointManager.stop();
```

Comment:
Do we know for sure that this method is only called once? If so, do we need to create the checkpoint manager during construction, or can we create it and close it within createResources?

Reply:
I also thought we could create an admin client only to create resources. But i think it is not that good that then we would create two kafka adminClients for the same system in the same manager, and also create another thread. I think it would be better if we only create the clients we really need to create.
I double checked the method is only called once. And advantages not to create it during construction?

Reply:
@MabelYC Sorry, don't understand the first part of your reply. What is that in response to?

Creating the checkpoint manager in the constructor and stopping it after the first use is :
1. Asymmetrical. Ideally this class should have lifecycle methods (start/stop) and the checkpoint manager would be stopped there. Also, looks like we don't even start the checkpoint manager?
2. Makes an unsafe assumption that the method is only called once. If it was ever called more than once, the second call would fail since the checkpoint manager is already stopped.

IMHO it will be cleaner to create a new checkpoint manager in createResources and close it after use. That way there is no additional overhead as long as it's called once, and its safe if its called more than once.

Reply:
Got it. Updated the pr. Created an adminClient for this function.